molecule_vagrant/modules/vagrant.py

```diff
@@ -254,22 +254,22 @@
     ##
     # Provider
     ##
-    c.vm.provider "{{ instance.provider }}" do |{{ instance.provider | lower }}, override|
+    c.vm.provider "{{ instance.provider }}" do |{{ instance.name | lower }}, override|
```

Comment:
Sorry, this looks wrong. It should be the provider name not the instance name.

Are you sure that the provider name is really ``vagrant-vmware-desktop`` ? The provider name is not always the plugin name and iirc, all vmware providers are starting with ``vmware_``. Having your instance started is possibly due to Vagrant being nice and ignore some parts of the ``Vagrantfile``.

Once the vm is created by vagrant, can you run ``vagrant global-status`` and check the provider names ?

Reply:
Hi Arnaud,
you are quite right. I didnt know about the `global-status` options, or at least hadnt anticipated that `name used to install provider != name used to referenced provider`
I have now changed the provider in `molecule.yml` to be `vmware_desktop` and all is well with your latest stable release. Ill close this PR. Thanks for the pointers.